framework_tvb/tvb/core/services/settings_service.py

```diff
@@ -237,3 +238,12 @@ def save_settings(self, **data):
             TvbProfile.current.manager.write_config_data(file_data)
             os.chmod(TvbProfile.current.TVB_CONFIG_FILE, 0o644)
         return anything_changed, first_run or db_changed
+
+    def check_tvb_folder(self, storage_path):
+        """
+        Check if the storage folder exists and is empty.
+        """
+        if storage_path:
+            if os.path.isdir(storage_path):
+                if os.listdir(storage_path):
+                    raise InvalidStorageException('TVB Storage should be empty, please set another folder.')
```

Comment:
Shouldn't this exception be raised only in case IS_FIRST_RUN ?
As I see from the unit-tests currently on any settings save this happens. 
We should be allowed to change other TVB settings later on.
Not sure if changing tvb_folder is allowed. If allowed, we should check that the NEW folder selected is empty.